### PR TITLE
Minor fixes that could go into 4.10 as well

### LINKED ISSUE #3496: Error in testbugfix with fr loaded
```
########> Diff in /home/gap/inst/gap-master/tst/testbugfix/2019-04-09-Lattice.\
tst:3
# Input is:
StructureDescription(g);;t:=TableOfMarks(g);;
# Expected output:
# But found:
Error, List Element: <list>[11] must have an assigned value
########
```

https://travis-ci.org/gap-infra/gap-docker-master-testsuite/builds/543426798
It also hangs `gap-master/tst/teststandard/twocohom.tst`

### PR DESCRIPTION
**Pull request by hulpke**
These are fixes for (an issue related to #3431) and #3496.
The circumstances of the errors are complicated and there were error messages, not wrong results, that I do not think they of help to describe them explicitly in an update description beyond summarizing them as "numerous errors")

Both issues are in 4.10 code.

Fixes #3496

### CODE REVIEW

**Comment by alex-konovalov** [comment]
@hulpke thanks. I've assigned this to 4.10.3 milestone - suggest to NOT to backport this to stable-4.10 until GAP 4.10.2 is out (expect to happen later this week).

File: lib/grppcfp.gi [hunk 1]
```diff
@@ -878,6 +878,9 @@ local g, sq, hom;
   g:=arg[1];
   sq:=CallFuncList(SQ,arg);
   hom:=GroupHomomorphismByImages(g,sq.image,GeneratorsOfGroup(g),sq.imgs);
+  if HasSize(g) and Size(g)=Size(sq.image) then
```
Comment (fingolfin) [inline lib/grppcfp.gi]:
```suggestion
  if HasSize(g) then
```

File: lib/grppcfp.gi [hunk 2]
```diff
@@ -878,6 +878,9 @@ local g, sq, hom;
...
+    SetIsInjective(hom,true);
```
Comment (fingolfin) [inline lib/grppcfp.gi]:
```suggestion
    SetIsInjective(hom, Size(g)=Size(sq.image));
```

**Comment by wilfwilson** [comment]
@alex-konovalov You can remove your requested changes.

**Comment by alex-konovalov** [comment]
@wilfwilson done!

**Comment by alex-konovalov** [comment]
@hulpke I set release notes: not needed" label then.

**Pull request merged**